File: 0461.Hamming Distance/bit.cpp
class Solution {
public:
    int hammingDistance(int a, int b) {
        unsigned int x = a;
        unsigned int y = b;
        int ret = 0;
        unsigned int mask = (1 << 31);
        for(int i = 0; i < 32; ++i){
            if((x & mask) != (y & mask)){
                ret++;
            }
            x = x << 1;
            y = y << 1;
        }
        return ret;
    }
};

File: 0461.Hamming Distance/bit.scala
object Solution {
    def hammingDistance(a: Int, b: Int): Int = {
        // x.asInstanceOf[unsignedInt];
        // y.asInstanceOf[unsignedInt];
        var x = a;
        var y = b;
        var mask: Int = (1 << 31);
        var ret = 0;
        for(i <- 0 until 32){
            if((x & mask) != (y & mask)){
                ret += 1;
            }
            x = x << 1;
            y = y << 1;
        }
        ret;
    }
}